src/twisted/web/_newserver.py

```diff
@@ -0,0 +1,257 @@
+            code=request.code,
+            length=request.sentLength or u"-",
+            referrer=referrer,
+            agent=agent,
+        )
+
+
+    # IProtocolNegotiationFactory
+    def acceptableProtocols(self):
+        """
+        Protocols this server can speak.
+        """
+        baseProtocols = [b'http/1.1']
+
+        if H2_ENABLED:
+            baseProtocols.insert(0, b'h2')
+
+        return baseProtocols
+
+
+
+def makeServer(resource, displayTracebacks=True, compressResponses=True,
+               timeout=_REQUEST_TIMEOUT, requestFactory=Request,
+               sessionFactory=Session, logger=None,
+               trustedReverseProxyIPs=None, reactor=None):
+    """
+    Create a web server.
+
+    @param resource: The resource to serve.
+    @type resource: L{twisted.web.resource.Resource}
+
+    @param displayTracebacks: Whether or not L{Resource}s will return
+        tracebacks experienced during processing to the web client.
+    @type displayTracebacks: L{bool}
+
+    @param compressResponses: Whether or not to compress responses delivered by
+        this L{server} with gzip.
+    @type compressResponses: L{bool}
+
+    @param timeout: The timeout for requests.
+    @type timeout: L{int}
+
+    @param requestFactory: A factory which is called with (channel) and creates
+        L{Request} instances.  Default to L{Request}.
+    @type requestFactory: C{callable} that returns objects that implement
+        L{twisted.web.iweb.IRequest}
+
+    @param sessionFactory: The Session class to use.  Default to L{Session}.
+    @type sessionFactory: L{Session}
+
+    @param logger: The logger for the server to use.
+    @type logger: L{twisted.logger.Logger}
+
+    @param trustedReverseProxyIPs: A C{list} of IPs which can send
+        C{X-Forwarded-For} headers that C{Request} will use as the IP, instead
+        of the origin IP.
+    @type trustedReverseProxyIPs: L{list} of L{str}
+
+    @param reactor: The C{Reactor} used for timekeeping.
+    """
+
+    if logger is None:
+        logger = Logger('twisted.web.server')
+
+    if reactor is None:
+        from twisted.internet import reactor
+
+    server = Server(protocol=_genericHTTPChannelProtocolFactory,
+                    displayTracebacks=displayTracebacks,
+                    compressResponses=compressResponses,
+                    trustedReverseProxyIPs=trustedReverseProxyIPs,
+                    timeout=timeout,
+                    resource=resource,
+                    requestFactory=requestFactory,
+                    sessionFactory=sessionFactory,
+                    logger=logger,
+                    reactor=reactor,
+    )
+
+    # Set up the Logger source correctly
+    logger.source = server
+    return server
+
+
+
+def makeCombinedLogFormatFileForServer(server, logFile,
+                                       logPublisher=globalLogPublisher):
+    """
+    Make a Combined Log Format file, logging from C{server}.
```

Comment:
This line does not add any new information, as is just the English version of makeCombinedLogFormatFileForServer :)